cypress/support/pages/admin_console/components/PageObject.ts

```diff
@@ -0,0 +1,320 @@
+export default class PageObject {
+  private selectItemSelectedIcon = ".pf-c-select__menu-item-icon";
+  private drpDwnMenuList = ".pf-c-dropdown__menu";
+  private drpDwnMenuItem = ".pf-c-dropdown__menu-item";
+  private drpDwnMenuToggleBtn = ".pf-c-dropdown__toggle";
+  private selectMenuList = ".pf-c-select__menu";
+  private selectMenuItem = ".pf-c-select__menu-item";
+  private selectMenuToggleBtn = ".pf-c-select__toggle";
+  private switchInput = ".pf-c-switch__input";
+  private formLabel = ".pf-c-form__label";
+  private chipGroup = ".pf-c-chip-group";
+  private chipGroupCloseBtn = ".pf-c-chip-group__close";
+  private chipItem = ".pf-c-chip-group__list-item";
+
+  protected assertExist(element: Cypress.Chainable<any>, exist: boolean) {
+    element.should((!exist ? "not." : "") + "exist");
+    return this;
+  }
+
+  protected assertIsVisible(
+    element: Cypress.Chainable<JQuery>,
+    isVisible: boolean
+  ) {
+    element.should((!isVisible ? "not." : "") + "be.visible");
+    return this;
+  }
+
+  protected assertIsEnabled(
+    element: Cypress.Chainable<JQuery>,
+    isEnabled: boolean = true
+  ) {
+    element.then(($btn) => {
+      if ($btn.hasClass("pf-m-disabled")) {
+        element.should(
+          (isEnabled ? "not." : "") + "have.class",
+          "pf-m-disabled"
+        );
+      } else {
+        element.should((isEnabled ? "not." : "") + "have.attr", "disabled");
+      }
+    });
+    return this;
+  }
+
+  protected assertIsDisabled(element: Cypress.Chainable<JQuery>) {
+    return this.assertIsEnabled(element, false);
+  }
+
+  protected assertHaveText(element: Cypress.Chainable<JQuery>, text: string) {
+    element.should("have.text", text);
+    return this;
+  }
+
+  protected assertHaveValue(element: Cypress.Chainable<JQuery>, value: string) {
+    element.should("have.value", value);
+    return this;
+  }
+
+  protected assertSwitchStateOn(
+    element?: Cypress.Chainable<JQuery>,
+    isOn: boolean = true
+  ) {
+    element = element ?? cy.get(this.switchInput);
+    element
+      .parent()
+      .contains(isOn ? "On" : "Off")
+      .should("be.visible");
+    return this;
+  }
+
+  protected assertSwitchStateOff(element?: Cypress.Chainable<JQuery>) {
+    return this.assertSwitchStateOn(element, false);
+  }
+
+  protected assertDropdownMenuIsOpen(
+    isOpen: boolean = true,
+    element?: Cypress.Chainable<JQuery>
+  ) {
+    element = element ?? cy.get(this.drpDwnMenuList);
+    this.assertExist(element, isOpen);
+    return this;
+  }
+
+  protected assertDropdownMenuIsClosed(element?: Cypress.Chainable<JQuery>) {
+    element = element ?? cy.get(this.drpDwnMenuList);
+    return this.assertDropdownMenuIsOpen(false, element);
```

Comment:
I would try to avoid re-assigning arguments here.
```suggestion
    return this.assertDropdownMenuIsOpen(false, element ?? cy.get(this.drpDwnMenuList));
```